front-end/src/components/charts/stock-timeline-chart.tsx

```diff
@@ -0,0 +1,227 @@
+  const stockTimeline = bridgeData(stockTimelineAbbreviated);
+
+  stockTimeline.forEach((entries, symbol) => {
+    const updatedEntries = entries.map((entry, index) => ({
+      ...entry,
+      index: index // Replace timestamp with index for even spacing
+    }));
+    stockTimeline.set(symbol, updatedEntries);
+  });
+
+
+  const [shownLines, setShownLines] = useState<Set<string>>(new Set(stockTimeline.keys()));
+
+  if(!timelineHasData(props.stockTimeline)) {
+    return  <Typography fontFamily="Montserrat" variant="h5">NO DATA</Typography>;
+  }
+
+  return (
+    <ResponsiveContainer width="100%" height="100%">
+      <LineChart margin={{ top: 5, right: 30, bottom: 80 }}>
+        <XAxis
+          type="number"
+          dataKey="index"
+          hide={true}
+        />
+        <YAxis dataKey="value" domain={['auto', 'auto']} />
+        <Tooltip content={<StockTimelineChartTooltip active={true} payload={[]} label={''} />} />
+        <Legend verticalAlign="top" height={50} onClick={onClickLegend(setShownLines)} />
+
+        {Array.from(stockTimeline.keys()).map((key) => (
+          <Line
+            key={key}
+            hide={!shownLines.has(key)}
+            type="monotone"
+            data={stockTimeline.get(key)}
+            dataKey="value"
+            name={key}
+            stroke={mapStringToColorCode(key)}
+            connectNulls={false}
+            isAnimationActive={true}
+            dot={true}
+          />
+        ))}
+      </LineChart>
+    </ResponsiveContainer>
+  );
+}
+
+function onClickLegend(setShownLines: React.Dispatch<React.SetStateAction<Set<string>>>) {
+  return function onClickLegend(...args: any[]): void {
+    if (args[0].type !== 'line' || !args[0].value) return;
+
+    const lineKey = args[0].value;
+    setShownLines((prevShownLines) => {
+      const updatedShownLines = new Set(prevShownLines);
+      if (updatedShownLines.has(lineKey)) {
+        updatedShownLines.delete(lineKey);
+      } else {
+        updatedShownLines.add(lineKey);
+      }
+      return updatedShownLines;
+    });
+  };
+}
+
+function timelineHasData(stockTimeline: Map<number, StockTimelineEntry[]>): boolean {
+  return stockTimeline.size >= 0 &&  Array.from(stockTimeline.values()).some(entries => entries.length > 0);
+}
+
+function mapStringToColorCode(input: string): string {
+  // Convert the input string to a hash
+  let hash = 0;
+  for (let i = 0; i < input.length; i++) {
+    hash = input.charCodeAt(i) + ((hash << 5) - hash);
+  }
+
+  // Use the hash to generate a hue value (0-360) in the HSL color space
+  const hue = Math.abs(hash % 12) * 30;
+  // Set saturation to 70% and lightness to 50% for good contrast with white text
+  const saturation = 70;
+  const lightness = 50;
+
+  // Convert HSL to RGB and then to hexadecimal color code
+  return hslToHex(hue, saturation, lightness);
+}
+
+// Helper function to convert HSL to Hex
+function hslToHex(h: number, s: number, l: number): string {
```

Comment:
Imma just trust this makes sense

Reply:
TLDR: this is code makes sure that there is good contract in the colors selected.